src/modifierBounty.ts

```diff
@@ -53,4 +56,31 @@ export default function registerBounty() {
       }
     }
   });
+}
+
+export function placeRandomBounty(bountyHunter: Unit.IUnit, underworld: Underworld, prediction: boolean, ignoreMax: boolean = false) {
+  let units = prediction ? underworld.unitsPrediction : underworld.units;
+
+  // Get existing bounty targets
+  const unitsWithBounty = getUnitsWithBounty(underworld, prediction);
+  // Max bounties = number of bounty hunters on team
+  const maxBounties = units.filter(u => u.faction == bountyHunter.faction && u.modifiers[bountyHunterId]).length;
+  if (ignoreMax || unitsWithBounty.length < maxBounties) {
+    // Find a random enemy unit and give it a bounty
+    // Unit must be alive, in enemy faction, not a doodad, and not yet have a bounty
+    units = units.filter(u => u.alive && (u.faction != bountyHunter.faction) && (u.unitSubType != UnitSubType.DOODAD) && !u.modifiers[bountyId]);
```

Comment:
Filter out `u.flaggedForRemoval` too.

This property is true when the unit is prepared to be removed.

Reply:
`alive` _probably_ covers this case anyway but it's probably safer to check for it despite

Reply:
Actually yeah we should definitely check for it.  It'll cover cases like when a unit is merged.  It never dies but one of them get's flaggedForRemoval and eventually removed buy may still exist transiently